src/configuration/CertificateDetailsModal.tsx

```diff
@@ -10,339 +8,307 @@ import {
-  const ExtensionKey: FC<ExtensionKeyProps> = ({ extension, extensionKey }) => {
-    //https://datatracker.ietf.org/doc/html/rfc5280
-    const getExtPropContent = (ext: any, key: string) => {
-      let result = '';
-      if (typeof ext[key] === 'object') {
-        result += JSON.stringify(ext[key]);
-      } else {
-        result += ext[key].toString();
-      }
-      return result;
-    };
-    if (extensionKey === 'name' || extensionKey === 'value') {
-      return <></>;
-    }
-    return (
-      <p style={{ textIndent: '25px' }}>
-        <b>{extensionKey}: </b>
-        <code>{getExtPropContent(extension, extensionKey)}</code>
-      </p>
-    );
-  };
-  interface ExtensionProps {
-    extension: any;
-    index: number;
-  }
+interface ExtensionsListProps {
+  crt: x509.X509Certificate;
+  order: number;
+}
 
-  const Extension: FC<ExtensionProps> = ({ extension: ext, index }) => {
-    return (
-      <>
-        <DescriptionList>
-          <DescriptionListGroup>
-            <DescriptionListTerm>
-              <b>
-                {index + 1} - {ext.name}
-              </b>
-            </DescriptionListTerm>
-          </DescriptionListGroup>
-        </DescriptionList>
-        {[
-          Object.keys(ext).map((key) => (
-            <ExtensionKey
-              extension={ext}
-              extensionKey={key}
-              key={'extension' + ext.name + key}
-            />
-          )),
-        ]}
-      </>
-    );
-  };
-  interface ExtensionsListProps {
-    crt: pki.Certificate;
-    order: number;
+//https://datatracker.ietf.org/doc/html/rfc5280
+interface ExtensionProps {
+  extension: x509.Extension;
+  index: number;
+}
+
+const Extension: FC<ExtensionProps> = ({ extension, index }) => {
+  const obj = extension.toTextObject();
+  let extTitle = extension.type;
+  if (obj[TextObject.NAME]) {
+    extTitle = obj[TextObject.NAME];
   }
 
-  const ExtensionsList: FC<ExtensionsListProps> = ({ crt, order }) => {
-    return (
-      <>
-        {crt.extensions.map((ext, index) => (
-          <Extension
-            extension={ext}
-            index={index}
-            key={'extenstions' + order}
-          />
-        ))}
-      </>
-    );
+  const getExtensionContent = () => {
+    const textObj = extension.toTextObject();
+    return TextConverter.serialize(textObj);
   };
 
-  const formatArray = (obj: Array<any>, indent = ''): string => {
-    let result = '';
-    const width = 6;
-    let wcnt = 0;
-    result += indent;
-    for (let i = 0; i < obj.length; i++) {
-      result += obj[i] + ',';
-      wcnt++;
-      if (wcnt === width) {
-        wcnt = 0;
-        result += '\n' + indent;
-      }
-    }
-    return result + '\n';
-  };
+  return (
+    <DescriptionListGroup>
+      <DescriptionListTerm>
+        <b>
+          {index + 1} - {extTitle}
+        </b>
+      </DescriptionListTerm>
+      <DescriptionListDescription>
+        <CodeBlockCode
+          style={{ backgroundColor: 'white' }}
```

Comment:
It's better to not manually style stuff in most cases. Because it'll have side effects elsewhere. For instance, setting the background white is what makes the code blocks unable to show the data appropriately in dark mode.

Reply:
ok, will remove the style. Thanks!

Reply:
@lavocatt I updated the PR (removed style and also improved the public key info display)